lib/rust/ensogl/core/src/display/object/class.rs

```diff
@@ -908,6 +1120,50 @@ pub trait ObjectOps<Host = Scene>: Object<Host> {
     }
 
 
+    // === Events ===
+
+    /// Emit a new event. See docs of [`event::Event`] to learn more.
+    fn emit_event<T>(&self, event: T)
+    where
+        Host: 'static,
+        T: 'static, {
+        self.display_object()._emit_event(event)
+    }
+
+    /// Create a new event handler for the given event type. See docs of [`event::Event`] to learn
+    /// more.
+    fn on_event<T>(&self) -> frp::Source<event::Event<Host, T>>
```

Comment:
Why we return `frp::Source` here? Do we encourage library users (devs) to emit such events by themselves? Maybe we should return `frp::Stream` instead?

I also don't like the term "Handler" in the docs. The handler is already handling the event: here we have only a stream, the handling still needs to be connected to.